File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.js
function solution(cards1, cards2, goal) {
    let answer = 'No';
    let [fivot1, fivot2, fivot3] = [0,0, 0]
    
    for(targetWord of goal){
        if (fivot1 < cards1.length && cards1[fivot1] === targetWord){
            fivot1 += 1
        }else if(fivot2 < cards2.length && cards2[fivot2] === targetWord){
            fivot2 += 1

        }   
    }
    answer  = (fivot1 + fivot2) === goal.length? "Yes" : "No" 
    return answer;
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        int pt1 = 0;
        int pt2 = 0;
        
        int goalPt = 0;
        
        while(pt1 < cards1.length || pt2 < cards2.length){
            if (goalPt >= goal.length){
                return "Yes";
            }
            
            String targetWord = goal[goalPt];
            
            if (pt1 < cards1.length && cards1[pt1].equals(targetWord)){
                pt1++;
                goalPt++;
            } else if (pt2 < cards2.length && cards2[pt2].equals(targetWord)){
                pt2++;
                goalPt++;
            } else{
                return "No";
            }
            
            
        }
        
        if (goalPt == goal.length){
            return "Yes";
        }
        
        return "No";
       
    }
}